modules/loader/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/db/Migration.scala

```diff
@@ -181,54 +179,39 @@ object Migration {
   def empty[F[_]: Applicative]: Migration[F] =
     Migration[F](Nil, Applicative[F].unit)
 
-  def buildBlock[F[_]: MonadThrow: DAO, I](description: Description, target: Target[I]): F[Option[Block]] =
+  def buildBlock[F[_]: MonadThrow: DAO, I](description: Description, target: Target[I]): F[List[Block]] =
     description match {
       case Description.Table(schemas) =>
-        val tableName = StringUtils.getTableName(schemas.latest)
-
-        val migrate: F[Option[Block]] = for {
-          schemaKey <- getVersion[F](tableName)
-          matches = schemas.latest.schemaKey == schemaKey
-          block <- if (matches) emptyBlock[F]
-                   else
-                     Control.getColumns[F](tableName).flatMap { (columns: List[ColumnName]) =>
-                       migrateTable(target, schemaKey, columns, schemas) match {
-                         case Left(migrationError) => MonadThrow[F].raiseError[Option[Block]](migrationError)
-                         case Right(block) => MonadThrow[F].pure(block.some)
-                       }
-                     }
-        } yield block
-
-        Control.tableExists[F](tableName).ifM(migrate, Monad[F].pure(target.createTable(schemas).some))
+        val schemaKeysWithModels = foldMapMergeRedshiftSchemas(schemas)
```

Comment:
Currently the return type of `foldMapMergedRedshiftSchemas` is `Map[SchemaKey, ShredModel]`.  In my opinion this is the wrong return type!  I would prefer it if that function returned something like `(ShredModel.GoodModel, Map[SchemaKey, ShredModel.RecoveryModel)`.  Because they it is explicit that that the function returns exactly one good model.

If we made that change in schema-ddl, then I think this code block here would be simpler.  For example, you wouldn't need to call `.get` on an option when searching for the good model.

Reply:
Having said that.....  you don't need to change the return type of `foldMapMergedRedshiftSchemas`.  I think there are other ways to make this current code block neater without changing schema-ddl.  I think the trick is to structure it like this:

```scala
val blocks: F[List[Block]] = schemaKeysWithModels.values.toList.traverseFilter {
  case gm: ShredModel.GoodModel =>
    ??? // something that returns F[Option[Block]]
  case rm: ShredModel.RecoveryModel =>
    ???
}
```

Reply:
I think it is better to fix this at the root rather than downstream, let me change the signature in schema-ddl